### ISSUE TITLE
False positive: `unused_variables` triggers if a binding is only used in a macro that doesn't use it in its expansion

### ISSUE THREAD

**Issue by CreepySkeleton** [rust-lang/rust-clippy#5472]
Code ([Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=32c63375879e55e2fc0bdc31610abea8)):
```rust
macro_rules! erase {
    ($name:ident) => ();
} 

fn foo(name: &str) {
    erase!(name);
}

fn main() { 
    foo("aaa")
}
```

After the macro has expanded, there's no mentioning of `name` being used anymore, despite the programmer *did* use it. This is especially annoying when the macro's expansion is empty only when certain `cfg`s are enabled, like `debug_assert!` is a noop in release builds.

Well, I'm not really sure it's possible to fix since clippy runs only after all the macros has been expanded.

**Comment by flip1995** [comment]
That is not a Clippy lint, but a rustc lint. These cases are one reason `_name` (underscore-bindings) exist, so you can explicitly mark bindings as "they are only used in special cases".
